src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1021,8 +1025,34 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     scan.Skip(method.Length);
                 }
 
-                scan.Take();
+                scan.Take(); // consume space
+
+                // begin consuming request-target
                 begin = scan;
+
+                var targetBegin = scan;
+
+                if (targetBegin.Peek() != ByteForwardSlash)
+                {
+                    string requestUriScheme;
+                    if (scan.GetKnownHttpSchema(out requestUriScheme))
```

Comment:
Yes, as Ben said, one goal was to minimize allocations. 

Also, AFAIK there are no **valid** scenarios using absolute-form and schemes other than `http://` and`https://`. Kestrel should reject requests like `GET otherscheme://localhost/ HTTP/1.1` as invalid.